src/WingetCreateCore/Common/GitHub.cs

```diff
@@ -406,6 +396,53 @@ private async Task<PullRequest> SubmitPRAsync(string packageId, string version,
             }
         }
 
+        private async Task<string> GetVersionDirectoryPath(string packageId, string version = null)
+        {
+            string appPath = Utils.GetAppManifestDirPath(packageId, string.Empty, '/');
+            var contents = await this.github.Repository.Content.GetAllContents(this.wingetRepoOwner, this.wingetRepo, appPath);
+            string directory;
+
+            if (string.IsNullOrEmpty(version))
+            {
+                // Get the latest version directory
+                directory = contents
+                    .Where(c => c.Type == ContentType.Dir)
+                    .OrderByDescending(c => c.Name, new VersionComparer())
+                    .Select(c => c.Path)
+                    .FirstOrDefault();
+            }
+            else
+            {
+                // Get the specified version directory
+                directory = contents
+                    .Where(c => c.Type == ContentType.Dir && c.Name.EqualsIC(version))
+                    .Select(c => c.Path)
+                    .FirstOrDefault();
+            }
+
+            return directory;
+        }
+
+        private async Task DeletePackageManifest(long forkRepoId, string packageId, string version, string branchName)
+        {
+            string versionDirectoryPath = await this.GetVersionDirectoryPath(packageId, version);
+
+            if (string.IsNullOrEmpty(versionDirectoryPath))
+            {
+                throw new NotFoundException(nameof(version), System.Net.HttpStatusCode.NotFound);
+            }
+
+            // Get all files in the version directory
+            var versionDirectoryContents = await this.github.Repository.Content.GetAllContents(forkRepoId, versionDirectoryPath);
+
+            // Delete files from the new branch in the forked repository
+            foreach (var file in versionDirectoryContents)

```

Comment:
Do we also need to handle cleaning up the folder of the old version manifests?

Reply:
No we don't. That is handled automatically by GitHub/Git (as git can track only "files" by default). To verify, you can see this in action with a test PR i made:

https://github.com/mdanish-kh/winget-pkgs/pull/65

master branch: https://github.com/mdanish-kh/winget-pkgs/tree/master/manifests/a/Anydo/Anydo
PR branch: https://github.com/mdanish-kh/winget-pkgs/tree/Anydo.Anydo-5.0.40-121efaa7-325a-4275-9680-20f8ae006913/manifests/a/Anydo/Anydo

The folder `5.0.38.17924` was automatically removed.